src/app_service/service/settings/service.nim

```diff
@@ -343,23 +343,17 @@ QtObject:
   proc getTelemetryServerUrl*(self: Service): string =
     return self.settings.telemetryServerUrl
 
-  proc saveSendStatusUpdates*(self: Service, value: bool): bool =
-    var newStatus = StatusType.Online
-    if not value:
-      # This looks unintuitive, but `StatusType.Offline` is interpreted
-      # as `StatusUpdate.UNKNOWN_STATUS_TYPE` in status-go, which causes an error.
-      # Hence `Invisible` (= 4 [which is `INACTIVE` in status-go]) is considerd offline.
-      newStatus = StatusType.Invisible
-
+  proc saveSendStatusUpdates*(self: Service, newStatus: StatusType): bool =
     try:
-      # The new user status needs to always be broadcast, so we need to update
-      # the settings accordingly and might turn it off afterwards (if user has 
-      # set status to "offline"
-      if (self.saveSetting(KEY_SEND_STATUS_UPDATES, true)):
-        let r = status_update.setUserStatus(int(newStatus))
-        if(self.saveSetting(KEY_SEND_STATUS_UPDATES, value)):
-          self.settings.sendStatusUpdates = value
-          return true
+      const propagateStatus = true
+      if(self.saveSetting(KEY_SEND_STATUS_UPDATES, propagateStatus) != true):
```

Comment:
Why is that needed? In the current flow the same settings update may happen a few lines below :thinking:

Reply:
We need to set it 2 times. Before status_update.setUserStatus it needs to be set alway to true. W/o this the database will not be updated in status_update.setUserStatus and the status change will not be propagated to other peers. After status_update.setUserStatus we need to set it to true only in case of AlwaysOnline or Automatic in order to send 5 minutes broadcast info to the peers.

Reply:
Correct. The comments in the removed code above try to elaborate on that.

@endulab might be worth keeping those comments as it otherwise causes confusion and we'll risk that someone will changes this code without knowing, which will cause regression (also in syncing)

Reply:
@PascalPrecht you're right. I just saw that you added this comment and fixed setting status somewhere in the middle of this task. I removed this during syncing, but I will revert it.